cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala

```diff
@@ -396,7 +396,7 @@ trait QueueBasedTraverser extends Traverser {
 /**
  * A traverser that keeps track of first depth of visiting a given node.
  */
-trait DepthTracker extends QueueBasedTraverser {
+trait DepthTracker[NodeType <: Node] extends QueueBasedTraverser[NodeType] {
```

Comment:
@szymonm while it technically does not have to be as you point out, there is a case for making it because the trait DepthTracker "is a" QueueBasedTraverser (because of 'extends' keyword) and it is a bit odd to not type parametrize it. e.g., here do we just want to have a QueueBasedTraverser of Node, and not of more specific NodeTypes? If so (and I am fine with it), your suggestion makes sense.